docs/docs/contribute/document-minimal-permissions.mdx

```diff
@@ -0,0 +1,223 @@
+
+## Detect minimal permissions
+
+To detect the minimal permissions for a command, we use [Dev Proxy](https://aka.ms/devproxy). Dev Proxy records the API requests that a command makes. Then, it analyzes the requests to determine the minimal permissions required for the command to run. For the analysis, it uses information about Microsoft Graph permissions from its documentation. For other APIs, such as SharePoint REST and CSOM, it uses information that we [curate together with the community](https://adoption.microsoft.com/sample-solution-gallery/sample/pnp-devproxy-sharepoint-api-minimal-permissions).
+
+To detect the minimal permissions for a command:
+
+### Start Dev Proxy
+
+<Tabs groupId="shell">
+  <TabItem label="PowerShell" value="powershell">
+    - Open a terminal.
+    - Change the working directory to the CLI for Microsoft 365 repo root.
+    - Start Dev Proxy with the permission detection preset:
+
+      ```PowerShell
+      npm run devproxy:permissions -- -- -e "tenant=<your_tenant>"
+      ```
+      
+      For example, if your tenant is `contoso.onmicrosoft.com` run:
+      ```PowerShell
+      npm run devproxy:permissions -- -- -e "tenant=contoso"
+      ```
+
+  </TabItem>
+  <TabItem label="zsh/bash" value="sh">
+    - Open a terminal.
+    - Change the working directory to the CLI for Microsoft 365 repo root.
+    - Start Dev Proxy with the permission detection preset:
+      ```bash
+      npm run devproxy:permissions -- -e="tenant=<your_tenant>"
+      ```
+      
+      For example, if your tenant is `contoso.onmicrosoft.com` run:
+      
+      ```bash
+      npm run devproxy:permissions -- -e="tenant=contoso"
+      ```
+
+  </TabItem>
+</Tabs>
+
+### Run the CLI command through proxy
+
+<Tabs groupId="shell">
+  <TabItem label="PowerShell" value="powershell">
+    - Open a new terminal.
+    - Set environment variables:
+
+      ```PowerShell
+      $env:HTTPS_PROXY="http://127.0.0.1:8000"
+      $env:NODE_TLS_REJECT_UNAUTHORIZED=0
+      ```
+
+    - Run the command:
+
+      ```PowerShell
+      m365 spo web list -u /
+      ```
+
+  </TabItem>
+  <TabItem label="zsh/bash" value="sh">
+    - Open a new terminal.
+    - Run the command:
+
+      ```bash
+      HTTPS_PROXY=http://127.0.0.1:8000 NODE_TLS_REJECT_UNAUTHORIZED=0 m365 spo web list -u /
+      ```
+
+  </TabItem>
+</Tabs>
+
+### Stop Dev Proxy recording
+
+- Press `s` in the Dev Proxy console.
+- Dev Proxy stops the recording and shows the results in the console.
```

Comment:
This isn't directly about this section, but more from my own experience. When testing with certain site addresses, I get a few warnings where it doesn’t even output the minimal permissions, it just says there’s no matching pattern, even though one is defined.

![image](https://github.com/user-attachments/assets/5594735a-551d-4c00-8598-41ef9ddd9ffe)
![image](https://github.com/user-attachments/assets/fcc09d81-5a4f-4cf4-8219-c02cb58957b6)

Reply:
If Dev Proxy doesn't find a matching pattern even though one is defined, that seems like a bug. Could you please log an issue with Dev Proxy so that we can double check all is working as intended?